### ISSUE TITLE
Can't get to work with Laravel Mix 3

### ISSUE THREAD

**Issue by gabrielkoerich** [inertiajs/inertia-vue#58]
Hi there! First of all, thanks for inertia. 

I'm trying to test it in my app and it works fine using laravel mix 4, but as I need dynamic imports and styles in .vue files, I tried to downgrade to mix 3 and got those errors:

```
ERROR in ./node_modules/inertia-vue/src/link.js
Module parse failed: Unexpected token (33:6)
You may need an appropriate loader to handle this file type.
|   render(h, { props, data, children }) {
|     return h('a', {
|       ...data,
|       attrs: {
|         ...data.attrs,
 @ ./node_modules/inertia-vue/src/index.js 3:0-32
 @ ./resources/inertia/inertia.js
 @ multi ./resources/inertia/inertia.js

ERROR in ./node_modules/inertia-vue/src/remember.js
Module parse failed: Unexpected token (34:63)
You may need an appropriate loader to handle this file type.
|       this.$watch(key, () => {
|         Inertia.remember(
|           this.$options.remember.data.reduce((data, key) => ({ ...data, [key]: this[key] }), {}),
|           stateKey
|         )
 @ ./node_modules/inertia-vue/src/app.js 3:0-33
 @ ./node_modules/inertia-vue/src/index.js
```

Not sure if I'm missing some rule or plugin. Any ideas?

Thanks.

**Comment by gwleuverink** [comment]
I have the same issue. The spread operator babel plugin is already included in mix so i'm at a loss what this could be.. @gabrielkoerich Did you manage to fix it?

**Comment by gwleuverink** [comment]
Found the issue I believe. It seems laravel mix does not transpile code imported from node modules. https://github.com/JeffreyWay/laravel-mix/issues/1805

A quick fix is to add the following to your webpack config. This will exclude transpilation of all node modules except inertia & inertia-vue:

``` JavaScript
mix.webpackConfig({
   module: {
      rules: [
         {
            test: /\.jsx?$/,
            exclude: /(node_modules\/(?!(inertia|inertia-vue))|bower_components)/,
            use: [
               {
                  loader: 'babel-loader',
                  options: Config.babel()
               }
            ]
         }
      ]
   }
})
```

A better solution in the long run might be to release upcoming versions of inertia & inertia-vue with transpiled source. I remember reading in one of the issues that that will be done when v1 is released.

@reinink Without the addition of the snippet above inertia will not work with laravel-mix v3. Should this be added to the readme?

**Comment by gwleuverink** [comment]
On a somewhat unrelated note. Anyone who wants to enable library code splitting in addition to dynamic imports can add this to webpack.mix.js:

``` JavaScript
const webpack = require('webpack')

mix.webpackConfig({
   plugins: [
      new webpack.optimize.CommonsChunkPlugin({
         name: '/js/vendors',
         minChunks: ({ resource }) => /node_modules/.test(resource),
      }),
      new webpack.optimize.CommonsChunkPlugin({
         name: '/js/manifest',
         minChunks: Infinity
      }),
   ]
});
```

This will create `manifest.js` and `vendors.js` files  just like mix.extract() in mix v4

**Comment by sebastiandedeyne** [comment]
This is solved with the release of v0.1.0

**Comment by roarkmccolgan** [comment]
Although closed, I just wanted to leave this here for anyone finding the same issues,
I had to downgrade laravel mix to v3 to enable multiple css and js files to prevent empty css files [See issue here](https://github.com/JeffreyWay/laravel-mix/issues/2055)

Also had to downgrade sass loader to v7 [see here](https://github.com/JeffreyWay/laravel-mix/issues/2206#issuecomment-527730992)
Then had to apply this fix 👆🏻(remove .default) to get it to work.

Using Inertia package versions below:
"@inertiajs/inertia": "^0.1.0",
"@inertiajs/inertia-vue": "^0.1.0",
